src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -576,10 +577,14 @@ protected function compileFor($expression)
      *
      * @param  string  $expression
      * @return string
+     *
+     * @throws \RuntimeException
      */
     protected function compileForeach($expression)
     {
-        preg_match('/\( *(.*) +as *([^\)]*)/i', $expression, $matches);
+        if (! preg_match('/\( *(.*) +as *([^\)]*)/i', $expression, $matches)) {
+            throw new RuntimeException("Invalid 'foreach' expression: ${expression}");
```

Comment:
`@expectedExceptionMessage Invalid 'foreach' expression: ($messages => $m)` says it does. Do you see any specific issue with it?